### PR TITLE
Extended guild classes

### LINKED ISSUE #215: [Feature Request] bring back extended guild selectors
**Describe the feature request**
yes

**What Plugin would it be for?**
yes

**Expected behavior**
yes

### PR DESCRIPTION
**Pull request by cloudrac3r**
Closes #215. This doesn't address the comment about guild names.

If you want to optimise out the ClassList thing into string matching functions then go ahead, but JS is fast so I don't care enough to do it myself.

The render function is called on a guild (and only that guild) when it is hovered or selected, which was very convenient for the creation of this plugin.

### CODE REVIEW

**Comment by cloudrac3r** [comment]
fuck you eslint i do what i want

File: src/Powercord/plugins/pc-guild-classes/index.js [hunk 1]
```diff
@@ -1,57 +1,56 @@
...
+
+  async _patchGuild () {
+    const guildInner = await waitFor('.blobContainer-239gwq');
+    const guildElement = guildInner.parentElement;
+    const Guild = getOwnerInstance(guildElement);
+    inject('cadence-guild-classes', Guild.constructor.prototype, 'render', (_, res) => {
+      if (res._owner) {
+        const { hovered } = res._owner.memoizedState;
+        const { selected } = res._owner.pendingProps;
+        const classList = new ClassList(res.props.className);
```
Comment (cyyynthia) [inline src/Powercord/plugins/pc-guild-classes/index.js]: is an extra class required? Wouldn't something like this enough?
```js
const classNames = [ ...new Set(className.split(' ')) ]
hovered && classNames.push('...')
selected && classNames.push('...')
res.props.className = classNames.join(' ');
```

File: src/Powercord/plugins/pc-guild-classes/index.js [hunk 2]
```diff
@@ -1,57 +1,56 @@
...
+  startPlugin () {
+    this._patchGuild();
+  }
+
+  pluginWillUnload () {
+    uninject('cadence-guild-classes');
+  }
+
+  async _patchGuild () {
+    const guildInner = await waitFor('.blobContainer-239gwq');
```
Comment (cyyynthia) [inline src/Powercord/plugins/pc-guild-classes/index.js]: Avoid using raw classes in here, fetch them dynamically

File: src/Powercord/plugins/pc-guild-classes/index.js [hunk 3]
```diff
@@ -1,57 +1,56 @@
...
+    const guildElement = guildInner.parentElement;
+    const Guild = getOwnerInstance(guildElement);
+    inject('cadence-guild-classes', Guild.constructor.prototype, 'render', (_, res) => {
```
Comment (cyyynthia) [inline src/Powercord/plugins/pc-guild-classes/index.js]: Keeping injection names in here with "pc-" would be more consistent as it's an internal plugin

File: src/Powercord/plugins/pc-guild-classes/index.js [hunk 4]
```diff
@@ -1,57 +1,56 @@
...
+      if (res._owner) {
+        const { hovered } = res._owner.memoizedState;
+        const { selected } = res._owner.pendingProps;
+        const classList = new ClassList(res.props.className);
+        classList.conditional(hovered, 'pgc-hovered');
```
Comment (cyyynthia) [inline src/Powercord/plugins/pc-guild-classes/index.js]: is a prefix really required here? wouldn't just go for .hovered be enough?

**Review by cyyynthia — CHANGES_REQUESTED** [review]

**Review by GriefMoDz — APPROVED** [review]

**Pull request merged**